### ISSUE TITLE
Re-add Multiblock structures

### ISSUE THREAD

**Issue by OpticFusion1** [OpticFusion1/Slimefun-Reloaded#3]
## General Troubleshooting
  
- [X] I have checked for similar issues.
- [X] I have updated to the latest version.
- [X] I have checked the branches or the maintainers' PRs for upcoming features/bug fixes.

**Is your feature request related to a problem? Please describe.**
N/A

**Describe the solution you'd like**
N/A

**Describe alternatives you've considered**
N/A

**Additional context**
Items are stored here https://github.com/TheBusyBiscuit/Slimefun4/blob/master/src/main/java/me/mrCookieSlime/Slimefun/utils/Utilities.java

More Multiblock related code here https://github.com/TheBusyBiscuit/Slimefun4/tree/master/src/main/java/me/mrCookieSlime/Slimefun/Objects and here https://github.com/TheBusyBiscuit/Slimefun4/tree/master/src/main/java/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/multiblocks and here https://github.com/TheBusyBiscuit/Slimefun4/tree/master/src/main/java/me/mrCookieSlime/Slimefun/Objects/handlers and here https://github.com/TheBusyBiscuit/Slimefun4/blob/master/src/main/java/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java

**Comment by OpticFusion1** [comment]
This is currently being worked on, but look at how shit the old version is 
![image](https://user-images.githubusercontent.com/37254722/69487061-b1ec8980-0e21-11ea-92f1-2dff0cfd2310.png)

**Comment by 2008Choco** [comment]
New potential system:
```java
    CuboidStructureBlueprint blueprint =
        StructureBlueprintFactory.create(new String[] { " N ", "WAW", " N " },
                                         new String[] { "EGE", "GGG", "EGE" },
                                         new String[] { "DDD", "DDD", "DDD" })
                                 .set('N', Material.OAK_LOG.createBlockData(b -> ((Orientable) b).setAxis(Axis.X)))
                                 .set('W', Material.OAK_LOG.createBlockData(b -> ((Orientable) b).setAxis(Axis.Z)))
                                 .set('A', Material.AIR).set('E', Material.EMERALD_BLOCK).set('G', Material.GOLD_BLOCK).set('D', Material.DIAMOND_BLOCK)
                                 .build();

    MultiBlock multiblock = new MultiBlock(new NamespacedKey(plugin, "MyMultiblock"), blueprint, BlockFace.NORTH);
```

Creates the following structure:
![structure](http://i.imgur.com/9jWuQ7A.png)

System may not be final and is subject to changes during implementation.

**Comment by OpticFusion1** [comment]
BUT it's still better than the current system >.>

**Comment by yannicklamprecht** [comment]
Would be cool to see that configurable. De-/Serializing Multiblock"recipes".

`
multiblock-blueprints:
    yourFancyPlugin:
        MyMultiblock:
            block-face: NORTH
            pattern:
               - [" N ", "WAW", " N "]
               - ["EGE", "GGG", "EGE"]
               -  ["DDD", "DDD", "DDD]
            blockData:
               N: 'minecraft:oak_log[axis=x]'
               W: 'minecraft:oak_log[axis=z]'
               A: 'minecraft:air'
               E: 'minecraft:emerald_block'
               G: 'minecraft:gold_block'
               D: 'minecraft:diamond_block'

`

**Comment by 2008Choco** [comment]
I see no real reason for this to be included. This multiblock API is primarily for internal use. Keep in mind that Slimefun has no reason to allow for multiblock structures to be configurable to the server owners. They should remain consistent from server to server. I have, however, considered including a way to load multiblock structures from files for developers so it doesn't necessarily have to be hard coded in source but rather just a file exported alongside the plugin.

**Comment by OpticFusion1** [comment]
> 
> 
> I see no real reason for this to be included. This multiblock API is primarily for internal use. Keep in mind that Slimefun has no reason to allow for multiblock structures to be configurable to the server owners. They should remain consistent from server to server. I have, however, considered including a way to load multiblock structures from files for developers so it doesn't necessarily have to be hard coded in source but rather just a file exported alongside the plugin.

Agreed, it being configurable shouldn't be a thing, however a way to load from files would be good. Would the structure just be a .txt or have its own file extension?

**Comment by yannicklamprecht** [comment]
> > I see no real reason for this to be included. This multiblock API is primarily for internal use. Keep in mind that Slimefun has no reason to allow for multiblock structures to be configurable to the server owners. They should remain consistent from server to server. I have, however, considered including a way to load multiblock structures from files for developers so it doesn't necessarily have to be hard coded in source but rather just a file exported alongside the plugin.
> 
> Agreed, it being configurable shouldn't be a thing, however a way to load from files would be good. Would the structure just be a .txt or have its own file extension?

I thought of a possibility of easily adding new multi block structures to the plugin by "config". Non user configurable, need to be pr'ed into the plugin or added as an moreMultiblockAddon.zip loaded by the plugin and each file in zip adds a new multi block.

**Comment by OpticFusion1** [comment]
Something like that multi-blocks are kind of code heavy, needs functionality ofc

**Comment by OpticFusion1** [comment]
Something like what you're suggesting should be an addon of its own

**Comment by 2008Choco** [comment]
> I thought of a possibility of easily adding new multi block structures to the plugin by "config"
While interesting, the user would really only have a way to define the structure of the multiblock but it would have no further functionality. That would require programming knowledge or some sort of scripting language integration which is a whole separate thing I really think is beyond the scope of the plugin. Meaning:
> Something like what you're suggesting should be an addon of its own
this seems like the correct approach. Perhaps that's something that could be worked on but as far as Slimefun is concerned, it has no place in the core of the project. Not to say it's a bad idea, it's just far beyond the goal :)

**Comment by OpticFusion1** [comment]
^
You're looking at skript complex, unless we turned it into something like

**Comment by 2008Choco** [comment]
This issue will remain idle until #5 has been completed, pull requested and merged. Multiblocks need a more solid API to work with before implementing the default machines.

**Comment by OpticFusion1** [comment]
> 
> 
>  Multiblocks need a more solid API

Lets be honest here, most things need a better API
